parquet/src/main/scala/magnolify/parquet/ParquetField.scala

```diff
@@ -366,6 +368,95 @@ object ParquetField {
     }
   }
 
+  private val KeyField = "key"
+  private val ValueField = "value"
+  private val KeyValueGroup = "key_value"
+  implicit def pfMap[K, V](implicit
+    pfKey: ParquetField[K],
+    pfValue: ParquetField[V]
+  ): ParquetField[Map[K, V]] = {
+    new ParquetField[Map[K, V]] {
+      override def buildSchema(cm: CaseMapper): Type = {
+        val keySchema = Schema.rename(pfKey.schema(cm), KeyField)
+        require(keySchema.isRepetition(Repetition.REQUIRED), "Map key must be required")
+        val valueSchema = Schema.rename(pfValue.schema(cm), ValueField)
+        val keyValue = Types
+          .repeatedGroup()
+          .addField(keySchema)
+          .addField(valueSchema)
+          .named(KeyValueGroup)
+        Types
+          .requiredGroup()
+          .addField(keyValue)
+          .as(LogicalTypeAnnotation.mapType())
+          .named("map")
+      }
+
+      override val hasAvroArray: Boolean = pfKey.hasAvroArray || pfValue.hasAvroArray
```

Comment:
I almost think we should throw an error if there's an `hasAvroArray` mismatch between key and value (and they both contain repeated types), since that probably points to a user error... that said, an iterable key is enough of an edge case it might not be worth implementing 😅